projects/ngx-maplibre-gl/src/lib/control/control.component.ts

```diff
@@ -65,20 +66,24 @@ export class ControlComponent<T extends IControl>
 
   control: T | CustomControl;
 
-  constructor(private mapService: MapService) {}
+  constructor(private mapService: MapService) {
+    afterNextRender(() => {
+      if (this.content.nativeElement.childNodes.length) {
+        this.control = new CustomControl(this.content.nativeElement);
+        this.mapService.mapCreated$.subscribe(() => {
+          this.mapService.addControl(this.control!, this.position);
+        });
+      }
+    })
+  }
 
   ngAfterContentInit() {
-    if (this.content.nativeElement.childNodes.length) {
-      this.control = new CustomControl(this.content.nativeElement);
-      this.mapService.mapCreated$.subscribe(() => {
-        this.mapService.addControl(this.control!, this.position);
-      });
-    }
+   
   }
 
   ngOnDestroy() {
-    if (this.mapService.mapInstance.hasControl(this.control)) {
-      this.mapService.removeControl(this.control);
-    }
+    // if (this.mapService.mapInstance.hasControl(this.control)) {
```

Comment:
I guess you could use "?" here so that if this is not initialized, then it won't do anything, can't you?
I'm not very worried about SSR memory leaks for this lib, right?

Reply:
I'm more concerned about leaks on the client actually, because the SSR apps today often soft navigation where possible, so they won't necessarily clean up the client until a proper page refresh is performed. I can see this when debugging the examples - I had to go into each one and only when i hit refresh they'd trigger some error.

Reply:
Hmm, yea I get your point here.
I'll see if I can look into it in the next few days...
Documentation is so lacking that it makes this whole process very painful...

Reply:
What about injecting the platform to check if its the browser and then do the clean up?

```ts
private readonly _isBrowser: boolean = inject(Platform).isBrowser;

ngOnDestroy() {
  if(this._isBrowser) {
    ...
  }
}
```